angular/index.d.ts

```diff
@@ -756,7 +756,7 @@ declare namespace angular {
          * @param comparator Function used to determine the relative order of value pairs.
          * @return An array containing the items from the specified collection, ordered by a comparator function based on the values computed using the expression predicate.
          */
-        <T>(array: T[], expression: string|((value: T) => any)|(((value: T) => any)|string)[], reverse?: boolean, comparator?: IFilterOrderByComparatorFunc): T[];
```

Comment:
However, the proposed approach doesn't work. See [playground](http://www.typescriptlang.org/play/#src=interface%20OrderBy%20%7B%0D%0A%20%20%20%20%3CT%2C%20A%20extends%20T%5B%5D%20%7C%20null%20%7C%20undefined%3E(%0D%0A%20%20%20%20%20%20%20%20array%3A%20A%2C%0D%0A%20%20%20%20%20%20%20%20expression%3F%3A%20string%20%7C%20((value%3A%20T)%20%3D%3E%20any)%20%7C%20(((value%3A%20T)%20%3D%3E%20any)%20%7C%20string)%5B%5D%2C%0D%0A%20%20%20%20%20%20%20%20reverse%3F%3A%20boolean%0D%0A%20%20%20%20)%3A%20A%3B%0D%0A%7D%0D%0A%0D%0Alet%20f%3A%20OrderBy%3B%0D%0A%0D%0Alet%20a%20%3D%20f(null%2C%20'z')%3B%20%2F%2F%20%60a%60%20is%20of%20type%20%60any%60%20not%20%60null%60%0D%0A).